Readability.js

```diff
@@ -1661,10 +1661,27 @@ Readability.prototype = {
         var input = node.getElementsByTagName("input").length;
 
         var embedCount = 0;
-        var embeds = node.getElementsByTagName("embed");
+        var videoCount = 0;
+        var embeds = this._concatNodeLists(
+          node.getElementsByTagName("object"),
+          node.getElementsByTagName("embed"),
+          node.getElementsByTagName("iframe"));
+
         for (var ei = 0, il = embeds.length; ei < il; ei += 1) {
-          if (!this.REGEXPS.videos.test(embeds[ei].src))
+          var attributeValues = [].map.call(node.attributes, function(attr) {
+            return attr.value;
+          }).join("|");
+
+          if (this.REGEXPS.videos.test(attributeValues) || this.REGEXPS.videos.test(node.innerHTML)) {
+            videoCount += 1;
+          } else {
             embedCount += 1;
```

Comment:
So now we're counting objects and iframes as embeds as well, which will cause the container to be removed when there are more than 1 (see checks below this code). I think that's probably fine but it's worth flagging up.

Reply:
I'm not sure if I follow. So, you mean we need to add new flag to specify whether we will :

- only count the `embed` tags (like before PR); or 
- count all `object`, `iframe` and `embed` (as in this PR) ?

If so, how should I name the flag ? Will `FLAG_COUNT_EMBED_ONLY` works ?

Reply:
I'm just noting that there's a side-effect to your patch that is quite different from the main goal of the patch, namely that if there's an arbitrary container node which contains 2 iframes that don't point to videos, the container will be removed.

I think that that side effect is probably OK; I don't think we need a separate flag for it. If we want to be more cautious, we could only increment `embedCount` if the node is really an `embed` node (ie not increment it for the iframe/object case).

Either way though, we don't need `videoCount`. If there is any video, we can return false immediately and stop all the loops. Because we never decrement `videoCount`, so when we return, for sure `videoCount` is still > 0, so we pass the condition below the loop:

```
        // If contains video, don't delete this node
        if (videoCount > 0) {
          return false;
        }
```

Reply:
> Either way though, we don't need `videoCount`

I've done it in [0a23f5](https://github.com/mozilla/readability/pull/526/commits/0a23f536326a5af02b7b10f5771cddfa82385217). Now, if the node contains an embedded video, the process will return false immediately and the `node` won't be deleted.